Fix docker environment

Badmuts opened the pull request:
This fixes the docker-compose environment. A `Makefile` was added to bootstrap the database. The database creation script was extended so the `MYSQL_USER` environment variable can used.

Badmuts committed:
fix: setup docker environment via makefile

Badmuts committed:
fix: nodemon is missing in admin container which crashes the server. Use the default command instead

Badmuts committed:
chore: update lock file

ToshKoevoets commented:
Hi Daan, deze fixes zijn om met docker te werken. Maar ik zou dat voor dev afraden. De huidige getting started gaat daar ook niet van uit.

Badmuts committed:
feat: add .nvmrc with node version 10 to setup environment

Badmuts committed:
feat: add setenv script to automate environment setup. cleanup docker-compose in favor of docker-compose.override.yml. add npm migration script

Badmuts committed:
docs: add
> openstad-app@0.9.0 db:install /Users/daanrosbergen/Code/openstad-app
> node install-databases.js

### 1. Create and fill tables API Database
### 2.1 Create tables Auth Database
Using environment: development

Knex: run
$ npm install mysql --save
Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
Error: Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Client_MySQL._driver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:31:12)
    at Client_MySQL.initializeDriver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:231:26)
    at Client_MySQL.Client (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:69:10)
    at new Client_MySQL (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:20:10)
    at Knex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js:53:28)
    at initKnex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js:57:10)

### 2.2 Fill tables Auth Database
Using environment: development

Knex: run
$ npm install mysql --save
Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
Error: Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Client_MySQL._driver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:31:12)
    at Client_MySQL.initializeDriver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:231:26)
    at Client_MySQL.Client (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:69:10)
    at new Client_MySQL (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:20:10)
    at Knex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js:53:28)
    at initKnex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js:57:10)

### 3.1 Create tables Image Database
Using environment: development

Knex: run
$ npm install mysql --save
Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
Error: Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Client_MySQL._driver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:31:12)
    at Client_MySQL.initializeDriver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:231:26)
    at Client_MySQL.Client (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:69:10)
    at new Client_MySQL (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:20:10)
    at Knex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js:53:28)
    at initKnex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js:57:10)

### 3.2 Fille tables Image Database
Using environment: development

Knex: run
$ npm install mysql --save
Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
Error: Cannot find module 'mysql'
Require stack:
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/index.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/knex.js
- /Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Client_MySQL._driver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:31:12)
    at Client_MySQL.initializeDriver (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:231:26)
    at Client_MySQL.Client (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/client.js:69:10)
    at new Client_MySQL (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/dialects/mysql/index.js:20:10)
    at Knex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/lib/knex.js:53:28)
    at initKnex (/Users/daanrosbergen/Code/openstad-app/node_modules/knex/bin/cli.js:57:10) step

Badmuts committed:
chore: update lock-file

Badmuts committed:
doc: add db:install step

Badmuts committed:
feat: add npm-run-all to run services with one npm script

Badmuts committed:
feat: create seperate env files to configure the services when using a node environment

Badmuts commented:
This PR now fixes both the docker and node environment 👍

ToshKoevoets commented:
Awesome, thanks for all contributions, will go through it ASAP, but looks
good.

Op wo 16 dec. 2020 om 12:07 schreef Daan Rosbergen <notifications@github.com
>:

> This PR now fixes both the docker and node environment 👍
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Amsterdam/openstad-app/pull/1#issuecomment-746114026>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJEFACPRTPEUWMQNCRL52LSVCIF5ANCNFSM4U4QX25A>
> .
>
